### PR TITLE
Implement presets

### LINKED ISSUE #266: Implement presets

### PR DESCRIPTION
**Pull request by dstallenberg**
Closes #266 
Depends on #260

### CODE REVIEW

File: libraries/base-testing-tool/lib/Configuration.ts [hunk 1]
```diff
@@ -301,15 +301,15 @@ export class Configuration {
...
           "total-time-budget": {
             alias: ["t"],
-            default: 3600,
+            default: Number.MAX_SAFE_INTEGER,
             description: "Total time budget",
             group: OptionGroups.Budget,
             hidden: false,
             type: "number",
           },
           "search-time-budget": {
```
[the remark this answers is not in the document]
Comment (mitchellolsthoorn) [reply libraries/base-testing-tool/lib/Configuration.ts:310]: Same applies to the other budgets

File: libraries/base-testing-tool/lib/presets/NSGAIIPreset.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Preset } from "@syntest/module";
+import { ArgumentsCamelCase } from "yargs";
+import { ArgumentsObject } from "../Configuration";
+
+export class NSGAIIPreset extends Preset {
```
Comment (mitchellolsthoorn) [inline libraries/base-testing-tool/lib/presets/NSGAIIPreset.ts:22]: Add paper reference comment

File: package-lock.json [hunk 3]
```diff
@@ -13146,6 +13145,47 @@
         "node": ">=10.24.0"
       }
     },
+    "presets/preset-dynamosa": {
```
Comment (mitchellolsthoorn) [inline package-lock.json:13148]: Rerun npm install

File: plugins/plugin-core-sfuzz/lib/SfuzzPreset.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ * limitations under the License.
+ */
+
+import { ArgumentsObject } from "@syntest/base-testing-tool";
+import { Preset } from "@syntest/module";
+import { ArgumentsCamelCase } from "yargs";
+
+export default class SfuzzPreset extends Preset {
+  constructor() {
+    super("Sfuzz", "Sfuzz preset");
```
Comment (mitchellolsthoorn) [inline plugins/plugin-core-sfuzz/lib/SfuzzPreset.ts:25]:
```suggestion
    super("sFuzz", "sFuzz preset");
```

File: plugins/plugin-core-sfuzz/lib/SfuzzPreset.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  modifyArgs(args: ArgumentsCamelCase): ArgumentsCamelCase {
+    (<ArgumentsObject>(<unknown>args)).searchAlgorithm = "Sfuzz";
+    (<ArgumentsObject>(<unknown>args)).objectiveManager = "Sfuzz";
```
Comment (mitchellolsthoorn) [inline plugins/plugin-core-sfuzz/lib/SfuzzPreset.ts:30]: Copy additionally settings from DynaMOSA

File: libraries/base-testing-tool/lib/Configuration.ts [hunk 6]
```diff
@@ -301,15 +301,15 @@ export class Configuration {
         .options({
           "total-time-budget": {
```
[the remark this answers is not in the document]
Comment (dstallenberg) [reply libraries/base-testing-tool/lib/Configuration.ts:302]: Fixed.

File: libraries/base-testing-tool/lib/plugins/SearchAlgorithmPlugin.ts [hunk 7]
```diff
@@ -23,12 +23,12 @@ import {
...
 import { Plugin } from "@syntest/module";
 import { PluginType } from "./PluginType";
-import { Offspring } from "@syntest/core";
+import { Procreation } from "@syntest/core";
 
 export type SearchAlgorithmOptions<T extends Encoding> = {
   objectiveManager: ObjectiveManager<T>;
   encodingSampler: EncodingSampler<T>;
-  offspring: Offspring<T>;
+  offspring: Procreation<T>;
```
[the remark this answers is not in the document]
Comment (dstallenberg) [reply libraries/base-testing-tool/lib/plugins/SearchAlgorithmPlugin.ts]: Fixed.

File: libraries/base-testing-tool/lib/presets/MOSAPreset.ts [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Preset } from "@syntest/module";
+import { ArgumentsCamelCase } from "yargs";
+import { ArgumentsObject } from "../Configuration";
+
+export class MOSAPreset extends Preset {
```
[the remark this answers is not in the document]
Comment (dstallenberg) [reply libraries/base-testing-tool/lib/presets/MOSAPreset.ts:22]: Fixed.

File: libraries/base-testing-tool/lib/presets/DynaMOSAPreset.ts [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Preset } from "@syntest/module";
+import { ArgumentsCamelCase } from "yargs";
+import { ArgumentsObject } from "../Configuration";
+
+export class DynaMOSAPreset extends Preset {
```
[the remark this answers is not in the document]
Comment (dstallenberg) [reply libraries/base-testing-tool/lib/presets/DynaMOSAPreset.ts:22]: Fixed.

File: package-lock.json (hunk 3 above)
Comment (dstallenberg) [reply package-lock.json]: FIxed.

**Review by mitchellolsthoorn — APPROVED** [review]

**Pull request merged**